From ed4a6478c7bbb2f03ea04c379bf6788a411c0243 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Thu, 11 Aug 2016 23:31:39 +0800
Subject: [PATCH 0454/1345] dts: a3700: add armada 37xx rz dtsi file

commit  f032a11d064a5905d3f1038c4ab631c58320f190 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Driver distinguish Z1 and A0 by whether the compatiable string is for
  RZ or not, we need to set RZ compatiable strings for Z1 fdt nodes
  which are different from A0;
- Add armada 37xx rz dtsi file inherited from armada 37xx dtsi
  file, thus the Z1 fdt nodes which are different from A0 can be
  overwritten with RZ compatiable strings in this file.

Change-Id: I8bf1eacdff600555cbd9943b451a086227d4db0b
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31902
Reviewed-by: Victor Gu <xigu@marvell.com>
Tested-by: Victor Gu <xigu@marvell.com>
(cherry picked from commit eaeb5ee3af66d5209c6703bf1b5093faa349ffe1)
Reviewed-on: http://vgitil04.il.marvell.com:8080/31868
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi |    2 +-
 arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi |    2 +-
 arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi |   51 +++++++++++++++++++++++
 3 files changed, 53 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi

diff --git a/arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi b/arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi
index bf2b9ea..30a4e57 100644
--- a/arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi
@@ -45,7 +45,7 @@
  *     OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include "armada-37xx.dtsi"
+#include "armada-37xx-rz.dtsi"
 
 / {
 	model = "Marvell Armada 3710 RZ SoC";
diff --git a/arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi b/arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi
index ceaca13..0bef261 100644
--- a/arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi
@@ -45,7 +45,7 @@
  *     OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include "armada-37xx.dtsi"
+#include "armada-37xx-rz.dtsi"
 
 / {
 	model = "Marvell Armada 3720 RZ SoC";
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi
new file mode 100644
index 0000000..8b4c98f
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi
@@ -0,0 +1,51 @@
+/*
+ * Device Tree Include file for Marvell Armada 37xx family of RZ SoCs.
+ *
+ * Copyright (C) 2016 Marvell
+ *
+ * Gregory CLEMENT <gregory.clement@free-electrons.com>
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This file is distributed in the hope that it will be useful
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "armada-37xx.dtsi"
+
+/ {
+	model = "Marvell Armada 37xx RZ SoC";
+};
-- 
1.7.9.5

